feat: query by tag

mamk95 opened the pull request:
This PR adds the ability to query by tag. You can use both tag ID or tag name/value as query, or a combination of ID and value. The query can be either a single string or a list of strings.

The Immich API requires a list of tag IDs. To be able to input a tag name, this suggested code first does a lookup of all tags using the Immich API, then translates the tag value to the tag ID.

Tested on Immich v1.126.1. Searching by tags was added in Immich v1.126.0, see this PR: https://github.com/immich-app/immich/pull/15395

Example:
```json
[
    {
        "name": "my album name",
        "query": {
            "tags": ["vacation1"]
        }
    },
    {
        "name": "my second album name",
        "query": {
            "tags": "545cf3a7-9f67-4028-9ed2-54bff6508052"
        }
    }
]
```

mamk95 committed:
Query by tag

kvalev commented:
Thanks for the PR! Can you also please update the version check here:

https://github.com/kvalev/immich-dynamic-albums/blob/223c1f50b7846bebca6fc274cd74ab9054d745bb/sync.py#L259-L260

Either flat out require 1.126.0 or if you want to be fancy, you can make the version check dynamic based on whether the user has configured tags or not.

mamk95 committed:
Minimum supported Immich version from v1.113.0 to v1.126.0

mamk95 committed:
README includes example for tags

mamk95 commented:
Changes since original PR:
- Rebased to include #4 
- Updated version check to require Immich v1.126.0 (up from v1.113.0)
- Updated the README to include an example for a tag query

Thanks for making this project. I've used it a lot. Really appreciate it!

kvalev commented:
Thanks! I created a new version - v0.2.0 which includes your changes, let me know if you have any problems with it.
